### PR TITLE
8330339: G1: Move some public methods to private in G1BlockOffsetTable APIs

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch moves several methods to the `private` part.

Thanks for taking the time to review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330339](https://bugs.openjdk.org/browse/JDK-8330339): G1: Move some public methods to private in G1BlockOffsetTable APIs (**Enhancement** - P4)

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18800/head:pull/18800` \
`$ git checkout pull/18800`

Update a local copy of the PR: \
`$ git checkout pull/18800` \
`$ git pull https://git.openjdk.org/jdk.git pull/18800/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18800`

View PR using the GUI difftool: \
`$ git pr show -t 18800`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18800.diff">https://git.openjdk.org/jdk/pull/18800.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18800#issuecomment-2059106678)

### CODE REVIEW

**Review by albertnetymk — COMMENTED** [review]
The inline.hpp should probably be updated as well to exclude private apis.

**Comment by lgxbslgx** [comment]
> The inline.hpp should probably be updated as well to exclude private apis.

Do you mean:

The methods `entry_for_addr` and `addr_for_entry` are not marked as `inline` in the `hpp`, so they should be moved to `cpp` and not `inlined`?

**Comment by albertnetymk** [comment]
Only `block_start_reaching_into_card` and methods called by it should be in inline.hpp.  All other methods can be moved to cpp.

**Comment by lgxbslgx** [comment]
I am confused by the `inline.hpp` file now. Previously, I think the non-trivial inlined methods should be put in `inline.hpp`. But now, it seems my thought is wrong. Now, I don’t really know when we should add such `inline.hpp` file and which methods should be put into `inline.hpp`. Could you explain more for me? Thanks.

**Comment by albertnetymk** [comment]
Since link-time-opt is disabled by default on most platforms (`# Only arm-32 should have link-time-opt enabled as default`), compiler needs to "see" the body of a method/function to not treat it as blackbox, i.e. emitting `call` in asm. As a result, inline.hpp is meant to expose some non-trivial & perf-critical methods to callers in another file (translation unit) so that compiler can perform more/better optimizations.

> I think the non-trivial inlined methods should be put in `inline.hpp`

I think it's more precise to say body of perf-critical public apis should be placed in hpp/inline.hpp. In order for compiler to see through those apis, their recursive callees should be "inlinable" (e.g. placed in inline.hpp) as well. Otherwise, the effect of placing those apis in inline.hpp will be limited.

**Comment by lgxbslgx** [comment]
Thanks for your explanation. I updated the code just now.

**Review by albertnetymk — APPROVED** [review]

**Comment by lgxbslgx** [comment]
@albertnetymk Thanks for your review. Waiting for another review.

**Review by walulyai — APPROVED** [review]

**Comment by lgxbslgx** [comment]
@albertnetymk @walulyai Thanks for your reviews. Integrating.

/integrate

**Pull request closed without merging**
